packages/block-directory/src/plugins/get-install-missing/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { createBlock, getBlockType } from '@wordpress/blocks';
+import { RawHTML } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { Warning } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import InstallButton from './install-button';
+
+const getInstallMissing = ( OriginalComponent ) => ( props ) => {
+	const { originalName, originalUndelimitedContent } = props.attributes;
+	// Disable reason: This is a valid component, but it's mistaken for a callback.
+	// eslint-disable-next-line react-hooks/rules-of-hooks
+	const { block, hasPermission } = useSelect(
+		( select ) => {
+			const { getDownloadableBlocks } = select( 'core/block-directory' );
+			const blocks = getDownloadableBlocks(
+				'block:' + originalName
+			).filter( ( { name } ) => originalName === name );
+			return {
+				hasPermission: select( 'core' ).canUser(
+					'read',
+					'block-directory/search'
+				),
+				block: blocks.length && blocks[ 0 ],
+			};
+		},
+		[ originalName ]
+	);
+
+	const { replaceBlock } = useDispatch( 'core/block-editor' );
+	const convertToHTML = () => {
+		replaceBlock(
+			props.clientId,
+			createBlock( 'core/html', {
+				content: originalUndelimitedContent,
+			} )
+		);
+	};
+
+	if ( ! hasPermission || ! block ) {
+		return <OriginalComponent { ...props } />;
+	}
+
+	const hasContent = !! originalUndelimitedContent;
+	const hasHTMLBlock = getBlockType( 'core/html' );
+
+	let messageHTML = sprintf(
+		/* translators: %s: block name */
+		__(
+			'Your site doesn’t include support for the %s block. You can try installing the block or remove it entirely.'
```

Comment:
It would be nice if we could add a button to remove the block, although not a blocker for this PR.

Reply:
Actually, my comment is with the original block, not this one. 

![Message](https://d.pr/i/EB4A0s.png)

Reply:
Yeah, it is following the pattern from the original core/missing block. I think the idea is that you'd remove it like any other block (click and hit delete, or use the toolbar menu to remove it), but you could open an issue about it if you think we need a button here.

Reply:
Note that this string interpolation [won't work for all locales](https://github.com/WordPress/gutenberg/pull/6354#issuecomment-384927905).

There isn't much else to do, especially since the `translators` note is already there. Some translations may work around their issues by rephrasing as e.g. `"%s: Your site doesn't include support for this block."`